### PR TITLE
Encourage improvements after submission

### PR DESCRIPTION
**Pull request by eToThePiIPower**
As per issue #77, the displays extra hints after the users submit the first iteration of each problem.

The hints encourage users on ways to refactor, and then to participate in the community by commenting on their submission.

Variation is added through randomizing the opening phrase to keep users from automatically ignoring the message. Additional and more creative phrases here would be appreciated.

### CODE REVIEW

File: cmd/submit.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+  "math/rand"
+  "time"
```
Comment (kytrinyx) [inline cmd/submit.go]: Would you make sure this is run through `gofmt`, please?

File: cmd/submit.go [hunk 2]
```diff
@@ -97,6 +99,28 @@ Your submission can be found online at %s
...
+
+* Remove as much duplication as you possibly can.
+* Optimize for readability, even if it means introducing duplication.
+* If you've removed all the duplication, do you have a lot of conditionals? Try
+  replacing the conditionals with polymorphism, if it applies in this language.
+  How readable is it?
+
+Then please share your thoughts in a comment on the submission. Did this
+experiment make the code better? Worse? Did you learn anything from it?
 `
```
Comment (kytrinyx) [inline cmd/submit.go]: Should we use a text/template for this?

File: cmd/submit.go [hunk 3]
```diff
@@ -97,6 +99,28 @@ Your submission can be found online at %s
...
+		msg += fmt.Sprintf("## %s", phrases[rand.Intn(len(phrases))])
+		msg += `
+
+Did you get the tests passing and the code clean? If you want to, these are some
+additional things you could try:
+
+* Remove as much duplication as you possibly can.
+* Optimize for readability, even if it means introducing duplication.
+* If you've removed all the duplication, do you have a lot of conditionals? Try
+  replacing the conditionals with polymorphism, if it applies in this language.
```
Comment (kytrinyx) [inline cmd/submit.go]: Let's not mention polymorphism. How about suggesting trying to get rid of conditionals without suggesting a tactic?

File: cmd/submit.go (hunk 2 above)
Comment (eToThePiIPower) [inline cmd/submit.go]: Looking into text/template, it seems like it'll make the code less readable overall. It either:
* Adds overhead for creating the template plus a data structure (struct/map), while still requiring all the text in the source file, or
* Creates a run-time dependency on an external file using `ParseFiles`, which doesn't really make sense.

If anything it makes sense to define the static text as a const definite at the end of the file, eg:
```
    msg += fmt.Sprintf("## %s", phrases[rand.Intn(len(phrases))])
    msg += tips
...
const tips = `
(tips here)
`
```

Comment (kytrinyx) [inline cmd/submit.go]: Yes, I think you're right. Defining it as a const would be better.

**Comment by kytrinyx** [comment]
Thanks!

In the future, after pushing a fix, would you please add a comment that it has been done? I don't get notifications when someone pushes to a PR.

**Comment by kytrinyx** [comment]
Now that this has been in production for a bit... what do you think?

**Comment by Tonkpils** [comment]
So, I've used it a bit since this was implemented and I feel that after
first or second submission it just becomes noise. It's good encouragement
initially, but I stopped reading it after my second submission.

**Comment by kytrinyx** [comment]
That's been my impression as well.

I think we should just take it back out, and keep working towards making the conversations on the site better.

**Pull request merged**